extrafiles/tooling/nktooling/__init__.py

```diff
@@ -0,0 +1,129 @@
+import os
+import glob
+import sys
+import tempfile
+
+READONLY = True
+
+def setup(args = None):
+	"""
+	Parses command-line arguments and sets module runtime flags.
+	"""
+	if args is None:
+		args = sys.argv
+
+	if "-h" in args or "--help" in args:
+		print("-h | --help    Print this message\n"
+			  "-w | --write   Apply fixes to source code file\n")
+		sys.exit(0)
+
+	global READONLY
+	READONLY = not ("-w" in args)
+
+def isReadonly():
+	"""
+	Returns whether module is in read-only mode (default: True).
+	In this case, FileRewriter.commit is silently deactivated.
+	"""
+	return READONLY
+
+def failIfReadonly(prgname):
+	"""
+	Helper function indented to give a summary after a test in read-only mode.
+	In read-only mode, fixes could not be applied, hence, we fail with a negative
+	return code. This can be used during CI to check whether repository is compliant.
+	"""
+	if not isReadonly():
+		return
+
+	print("Changes are necessary.")
+	print("Use '%s -w' to apply changes." % prgname)
+	sys.exit(-1)
```

Comment:
Traditionally, UNIX programs exit with 1 on error.

Reply:
Technically `-1` is at least 32 times stronger than `1` ;) But you're right. I'll fix it.